iml-agent/src/action_plugins/stratagem/action_purge.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, fidlist, http_comms::mailbox_client};
+use futures::{Future, Stream};
+
+pub fn purge_files(
+    device: &String,
+    args: impl IntoIterator<Item = String>,
+) -> Result<(), ImlAgentError> {
+    liblustreapi::rmfids(&device, args).map_err(ImlAgentError::LiblustreError)
+}
+
+pub fn read_mailbox(device: &str, mailbox: &str) -> impl Future<Item = (), Error = ImlAgentError> {
+    let mntpt = match liblustreapi::search_rootpath(&device) {
+        Ok(m) => m,
+        Err(e) => {
+            panic!("Failed to find rootpath({}) -> {:?}", device, e);
+            //return futures::future::err(ImlAgentError::LiblustreError(e));
```

Comment:
Can this be removed?

Reply:
I'd like to return a future::err.. but I get type errors.